### PR TITLE
Fix package links in new documentation generator.

### PR DESCRIPTION
**Pull request by cgranade**
This PR fixes the "Package" link that appears in the output from the new Markdown-based documentation tool. In particular, this PR fixes a typo in Sdk.targets that caused the package ID to not be passed to the rewrite step correctly, and adds a check for `IsNullOrWhitespace` to ensure that empty package IDs are treated the same way as `null`.

### CODE REVIEW

**Review by samarsha — APPROVED** [review]

**Comment by cgranade** [comment]
Looks like CI failed due to a temporary outage with a required NuGet feed. Will rerun.

**Comment by cgranade** [comment]
@samarsha: Thanks for the review! Would you be willing to take a quick look again? @msoeken brought another minor issue to my awareness such that I modified the PR to address that as well. Thanks!

File: src/Documentation/DocumentationGenerator/Extensions.cs [hunk 1]
```diff
@@ -179,11 +179,37 @@ TypedExpression input
...
-            SyntaxTreeToQsharp.DeclarationSignature(
+        internal static string ToSyntax(this QsFunctor functor) =>
+            functor.Tag switch
+            {
+                QsFunctor.Tags.Adjoint => "Adj",
+                QsFunctor.Tags.Controlled => "Ctl",
+                _ => "__invalid__"
+            };
+
+        internal static string ToSyntax(this QsCallable callable, Action<IRewriteStep.Diagnostic>? onDiagnostic = null)
```
Comment (samarsha) [inline src/Documentation/DocumentationGenerator/Extensions.cs:190]: `onDiagnostic` is an unused parameter?

File: src/Documentation/DocumentationGenerator/Extensions.cs [hunk 2]
```diff
@@ -179,11 +179,37 @@ TypedExpression input
...
+        {
+            var signature = SyntaxTreeToQsharp.DeclarationSignature(
                 callable, SyntaxTreeToQsharp.Default.ToCode
             );
```
Comment (samarsha) [inline src/Documentation/DocumentationGenerator/Extensions.cs:194]:
```suggestion
                callable, SyntaxTreeToQsharp.Default.ToCode);
```

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1111.md

Looks like there are a lot of StyleCop violations in this file, actually. :) Out of curiosity, are the squigglies showing up for you? It should look like:

image

File: src/Documentation/DocumentationGenerator/Extensions.cs [hunk 3]
```diff
@@ -179,11 +179,37 @@ TypedExpression input
...
+            // the characteristics (e.g.: `is Adj + Ctl`) for the callable, so
+            // we add that here.
+            var characteristics = callable.Signature.Information.Characteristics.SupportedFunctors switch
+            {
+                var nullable when nullable.IsNull => "",
+                { Item: var item } => item switch
+                {
+                    { Count: 0 } => "",
+                    // Be sure to add the leading space before is!
+                    _ => $" is {string.Join("+", item.Select(functor => functor.ToSyntax()))}"
```
Comment (samarsha) [inline src/Documentation/DocumentationGenerator/Extensions.cs:206]: Is there not supposed to be a space around `+` in the documentation?

File: src/Documentation/DocumentationGenerator/Extensions.cs [hunk 4]
```diff
@@ -179,11 +179,37 @@ TypedExpression input
...
+                }
```
Comment (samarsha) [inline src/Documentation/DocumentationGenerator/Extensions.cs:207]:
```suggestion
                { Item: { Count: 0 } } => "",
                // Be sure to add the leading space before is!
                { Item: var functors} => $" is {string.Join("+", functors.Select(functor => functor.ToSyntax()))}"
```

**Review by samarsha — APPROVED** [review]

File: src/Documentation/DocumentationGenerator/Extensions.cs [hunk 5]
```diff
@@ -179,11 +179,37 @@ TypedExpression input
...
+            var signature = SyntaxTreeToQsharp.DeclarationSignature(
                 callable, SyntaxTreeToQsharp.Default.ToCode
             );
 
+            // The signature provided by SyntaxTreeToQsharp doesn't include
+            // the characteristics (e.g.: `is Adj + Ctl`) for the callable, so
+            // we add that here.
+            var characteristics = callable.Signature.Information.Characteristics.SupportedFunctors switch
+            {
+                var nullable when nullable.IsNull => "",
```
Comment (samarsha) [inline src/Documentation/DocumentationGenerator/Extensions.cs:201]:
```suggestion
                { IsNull: true } => "",
```

...maybe. :)

File: src/Documentation/DocumentationGenerator/Extensions.cs (hunk 3 above)
Comment (cgranade) [reply src/Documentation/DocumentationGenerator/Extensions.cs:206]: Good catch, thank you!
```suggestion
                    _ => $" is {string.Join(" + ", item.Select(functor => functor.ToSyntax()))}"
```

File: src/Documentation/DocumentationGenerator/Extensions.cs (hunk 4 above)
Comment (cgranade) [reply src/Documentation/DocumentationGenerator/Extensions.cs]: For some reason I thought that nested pattern was a C# 9 feature... thanks for the suggestion!

File: src/Documentation/DocumentationGenerator/Extensions.cs (hunk 2 above)
Comment (cgranade) [reply src/Documentation/DocumentationGenerator/Extensions.cs:194]: I'll admit, I find code that follows SA1111 _much_ more difficult to read... maybe I've spent too long in PEP 8 codebases, which requires the exact opposite. In any case, thanks for raising that, will go on and fix.

File: src/Documentation/DocumentationGenerator/Extensions.cs (hunk 1 above)
Comment (cgranade) [reply src/Documentation/DocumentationGenerator/Extensions.cs:190]: Good catch, thank you; it used to be used, but that was stripped out in response to one of @bettinaheim's earlier points of feedback; must have missed a case.

File: src/Documentation/DocumentationGenerator/Extensions.cs (hunk 4 above)
Comment (samarsha) [reply src/Documentation/DocumentationGenerator/Extensions.cs]: Looks like this change was not applied - did nested patterns turn out to be a C# 9 feature? (It worked in my IDE but I didn't try building it.)

Comment (cgranade) [reply src/Documentation/DocumentationGenerator/Extensions.cs]: Ah, sorry, the web UI wouldn't allow me to accept the suggestion so I did it locally then forgot to push. Fixed now.

**Comment by cgranade** [comment]
It looks like the e2e test failure is in an unrelated component; are builds working at the moment?

```
       "/home/vsts/work/1/s/drop/Tests/tests.generated/tests-all/tests-all.csproj" (VSTest target) (1:7) ->
       "/home/vsts/work/1/s/drop/Tests/tests.generated/tests-all/tests-all.csproj" (default target) (1:8) ->
       (CoreCompile target) -> 
         CoreTests.cs(43,24): error CS0246: The type or namespace name 'AbstractFactory<>' could not be found (are you missing a using directive or an assembly reference?) [/home/vsts/work/1/s/drop/Tests/tests.generated/tests-all/tests-all.csproj]
```

**Comment by rmshaffer** [comment]
> It looks like the e2e test failure is in an unrelated component; are builds working at the moment?

No, I believe the E2E build is broken at the moment until https://github.com/microsoft/qdk/pull/27 is merged.

**Pull request merged**